src/drivers/intel/cavs/dmic.c

```diff
@@ -1489,8 +1485,18 @@ static int dmic_probe(struct dai *dai)
 	dmic->state = COMP_STATE_READY;
 
 	/* register our IRQ handler */
-	interrupt_register(dmic_irq(dai), IRQ_AUTO_UNMASK, dmic_irq_handler,
-			   dai);
+	ret = interrupt_register(dmic_irq(dai), IRQ_AUTO_UNMASK,
```

Comment:
same above, why dmic = rzalloc isn't tested is weird

Reply:
There's a check for dmic == NULL after rzalloc() but the part of code needed to be expanded  in Github view. Did you mean that?